File: backend/src/main/java/com/scim/service/LocationService.java
package com.scim.service;

import com.scim.dto.request.LocationRequest;
import com.scim.dto.response.LocationNodeResponse;
import com.scim.dto.response.LocationResponse;

import java.util.List;

public interface LocationService {

    List<LocationResponse> getByWarehouseId(Long warehouseId);

    List<LocationNodeResponse> getWarehouseLocationTree(Long warehouseId);

    LocationResponse getById(Long id);

    LocationResponse create(LocationRequest request);

    LocationResponse update(Long id, LocationRequest request);

    void delete(Long id);
}

File: frontend/src/services/locationService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  LocationResponse, 
  LocationRequest 
} from '@/types';

export const locationService = {
  getById: (id: number): Promise<ApiResponse<LocationResponse>> => {
    return apiClient.get(`/locations/${id}`);
  },

  create: (data: LocationRequest): Promise<ApiResponse<LocationResponse>> => {
    return apiClient.post('/locations', data);
  },

  update: (id: number, data: LocationRequest): Promise<ApiResponse<LocationResponse>> => {
    return apiClient.put(`/locations/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/locations/${id}`);
  },

  getByWarehouse: (warehouseId: number): Promise<ApiResponse<LocationResponse[]>> => {
    return apiClient.get(`/locations/warehouse/${warehouseId}`);
  },
};
